[a11y] fix VoiceOver navigation

thedrick opened the pull request:
Thanks for submitting a PR! Please read these instructions carefully:

- [ ] Explain the **motivation** for making this change.
- [ ] Provide a **test plan** demonstrating that the code is solid.
- [ ] Match the **code formatting** of the rest of the codebase.
- [ ] Target the `master` branch, NOT a "stable" branch.

## Motivation (required)

This commit disables removeClippedSubviews for all react views when VoiceOver is enabled on iOS. The issue is that VoiceOver relys on the view hierarchy to know how to navigate between elements when the user swipes left and right. When removeClippedSubviews is enabled, all views
offscreen are removed from the view hierarchy, causing VoiceOver to stop navigating when it gets to the last view rendered on screen. This PR fixes that by forcing removeClippedSubviews to be false for VoiceOver users.

## Test Plan (required)

Turned on VoiceOver and verified that navigation worked again with these changes.

## Next Steps

Sign the [CLA][2], if you haven't already.

Small pull requests are much easier to review and more likely to get merged. Make sure the PR does only one thing, otherwise please split it.

Make sure all **tests pass** on both [Travis][3] and [Circle CI][4]. PRs that break tests are unlikely to be merged.

For more info, see the ["Pull Requests"][5] section of our "Contributing" guidelines.

[1]: https://medium.com/@martinkonicek/what-is-a-test-plan-8bfc840ec171#.y9lcuqqi9
[2]: https://code.facebook.com/cla
[3]: https://travis-ci.org/facebook/react-native
[4]: http://circleci.com/gh/facebook/react-native
[5]: https://github.com/facebook/react-native/blob/master/CONTRIBUTING.md#pull-requests

thedrick committed:
[a11y] fix VoiceOver navigation

This commit disables removeClippedSubviews for all react views when VoiceOver is enabled on iOS. The issue is that VoiceOver relys on the view hierarchy to know how to navigate between elements when the user swipes left and right. When removeClippedSubviews is enabled, all views
offscreen are removed from the view hierarchy, causing VoiceOver to stop navigating when it gets to the last view rendered on screen. This PR fixes that by forcing removeClippedSubviews to be false for VoiceOver users.

thedrick committed:
[a11y] fix VoiceOver navigation

This commit disables removeClippedSubviews for all react views when VoiceOver is enabled on iOS. The issue is that VoiceOver relys on the view hierarchy to know how to navigate between elements when the user swipes left and right. When removeClippedSubviews is enabled, all views
offscreen are removed from the view hierarchy, causing VoiceOver to stop navigating when it gets to the last view rendered on screen. This PR fixes that by forcing removeClippedSubviews to be false for VoiceOver users.

thedrick committed:
Merge branch 'thedrick-voiceover-fixes' of https://github.com/airbnb/react-native into thedrick-voiceover-fixes

thedrick committed:
Merge branch 'thedrick-voiceover-fixes' of https://github.com/airbnb/react-native into thedrick-voiceover-fixes

lelandrichardson commented on React/Views/RCTView.m:
```diff
@@ -118,6 +118,11 @@ - (instancetype)initWithFrame:(CGRect)frame
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(voiceOverStatusDidChange:)
+                                                 name:UIAccessibilityVoiceOverStatusChanged
+                                               object:nil];
```
i'm a little bit concerned about adding a new observer for *every* RCTView instantiated. That could be thousands of views. Can we figure out a way to do this with only a single observer?

thedrick replied:
that's a great point, I'll investigate a solution that removes the dependency on NSNotificationCenter. Though, I can't imagine there would be thousands of views instantiated at one time in an application - if that were the case this one observer wouldn't be the issue, it would be the memory bloat from the views themselves.

lelandrichardson replied:
RN produces quite large view hierarchies. thousands might be a little aggressive, but hundreds definitely happens quite often.

thedrick committed:
Merge branch 'thedrick-voiceover-fixes' of https://github.com/airbnb/react-native into thedrick-voiceover-fixes

thedrick commented:
@lelandrichardson I updated the PR and removed the NSNotificationCenter calls. I feel weird about resetting a variable inside the setter method... but I couldn't find another workaround to get the views to behave properly with and without VO running. Let me know if you have other thoughts on how to accomplish this cleanly

lelandrichardson commented:
@thedrick LGTM. You can go ahead and merge this, and then gabe or myself can help you through the version bump + upgrade process
